### PR TITLE
feat: `select {alias = this}`

### PR DESCRIPTION
**Pull request by aljazerzen**

### CODE REVIEW

File: prqlc/prqlc/tests/integration/sql.rs [hunk 1]
```diff
@@ -4977,3 +4977,18 @@ fn test_param_declarations() {
       b = $a
     "###);
 }
+
+#[test]
+fn test_relation_aliasing() {
+    assert_display_snapshot!(compile(
+        r###"
+    from x | select {y = this} | select {y.hello}
```
Comment (max-sixty) [inline prqlc/prqlc/tests/integration/sql.rs:4985]: Cool! So `y` is a tuple / relation

**Review by max-sixty — APPROVED** [review]

File: prqlc/prqlc/tests/integration/sql.rs (hunk 1 above)
Comment (aljazerzen) [reply prqlc/prqlc/tests/integration/sql.rs:4985]: Idea is that if `x` is of type `[{a = int, b = bool}]`, then
```
x | select {y = this}
```
... will have the type of `[{y = {a = int, b = bool}}]`.

... which then gets flattened when converting to an SQL relation.

Comment (max-sixty) [reply prqlc/prqlc/tests/integration/sql.rs:4985]: Yes perfect.

Will be great to have an LSP one day, would be so cool to hover over `y` and get that type...

**Comment by vanillajonathan** [comment]
Maybe you can add something to the changelog, or some release notes or to the documentation something about this.

**Comment by aljazerzen** [comment]
Maybe, but it is a minor change, implemented with a hack. I plan to strip the "`Lineage`" part of our resolver out, and the new implementation will have this feature. So I've just added this hack to the lineage so the breakage between the new and the old implementations will be as small as possible.

So let's add this change to the changelog when the new type-based resolver is merged.

**Pull request merged**